Week3/homework/step2-7.js

```diff
@@ -20,4 +24,7 @@ f2(y);
 
 console.log(y);
 
-// Add your explanation as a comment here
+// As Jash explained in this and earlier lessons,
+// 1 st snippet writes 9 and 2n d writes { x: 10,}. JS calls by value, but in object and other complex data types with reference.
+// we used 9 and made val value 10, but it did not change x variable 's value. Complex data types such as object take place in memory.
+// So we modified the value of const y which has place in the memory,reference. So we are manipulating the memory place.
```

Comment:
Yep, good explanation! Sounds like you've got a good grasp on primitive vs complex types.